arches_resource_sets/views/resource_set_view.py

```diff
@@ -0,0 +1,50 @@
+import uuid
+
+from django.http import Http404
+
+from arches.app.views.api import APIBase
+from arches_resource_sets.models import ResourceSet
+from arches.app.utils.response import JSONErrorResponse, JSONResponse
+from arches.app.utils.betterJSONSerializer import JSONDeserializer, JSONSerializer
+
+
+class ResourceSetView(APIBase):
+    def get(self, request, set_id=None):
+        if set_id is None:
+            return JSONResponse({"resource_sets": ResourceSet.objects.all()})
+        set = ResourceSet.objects.filter(id=set_id).values().first()
+        return JSONResponse({"resource_set": set})
+
+    def put(self, request, set_id):
+        try:
+            set = ResourceSet.objects.get(id=set_id)
+        except ResourceSet.DoesNotExist:
+            return JSONErrorResponse("Could not update resource set", "Resource set id '{}' not found".format(set_id), status=404)
+        
+        request_body = JSONDeserializer().deserialize(request.body)
+        description = request_body["description"] if "description" else ""
+        set.description = description
+        set.save()
+
+        return JSONResponse({"resource_set": {"id": str(set_id)}})
+
```

Comment:
I think if a user tries to update the resource set but didn't include a body or for whatever reason something else fails, we shouldn't do anything

```suggestion
        try:
            request_body = JSONDeserializer().deserialize(request.body)
            description = request_body["description"]
            set.description = description
            set.save()
        except:
            pass
```

Reply:
This seems more like it should return an error, a 400 specifically.  If a user submits a PUT without a body we'll still have to return a status code.   200 seems wrong - they didn't actually PUT anything, and nothing was saved.  Returning the object unchanged seems wrong, since that should be a GET request, not a PUT.  Deleting the object outright (since there is no object posted) seems wrong - they should be issuing a DELETE.

Reply:
Yeah returning a 400 error makes more sense.